In [2]:
import pandas as pd
import plotly.express as px

px.defaults.width = 800                 
px.defaults.height = 500                

In [3]:
df_tinder = pd.read_csv("../data/speed_dating_data.csv", encoding='cp1252')

print(df_tinder.shape)
df_tinder.head()

(8378, 195)


,iid,id,gender,idg,condtn,wave,round,position,positin1,order,...,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
4,1,1.0,0,1,1,1,10,7,NaN,7,...,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN


In [4]:
with pd.option_context("display.max_columns", None):
    display(df_tinder.head())

,iid,id,gender,idg,condtn,wave,round,position,positin1,order,partner,pid,match,int_corr,samerace,age_o,race_o,pf_o_att,pf_o_sin,pf_o_int,pf_o_fun,pf_o_amb,pf_o_sha,dec_o,attr_o,sinc_o,intel_o,fun_o,amb_o,shar_o,like_o,prob_o,met_o,age,field,field_cd,undergra,mn_sat,tuition,race,imprace,imprelig,from,zipcode,income,goal,date,go_out,career,career_c,sports,tvsports,exercise,dining,museums,art,hiking,gaming,clubbing,reading,tv,theater,movies,concerts,music,shopping,yoga,exphappy,expnum,attr1_1,sinc1_1,intel1_1,fun1_1,amb1_1,shar1_1,attr4_1,sinc4_1,intel4_1,fun4_1,amb4_1,shar4_1,attr2_1,sinc2_1,intel2_1,fun2_1,amb2_1,shar2_1,attr3_1,sinc3_1,fun3_1,intel3_1,amb3_1,attr5_1,sinc5_1,intel5_1,fun5_1,amb5_1,dec,attr,sinc,intel,fun,amb,shar,like,prob,met,match_es,attr1_s,sinc1_s,intel1_s,fun1_s,amb1_s,shar1_s,attr3_s,sinc3_s,intel3_s,fun3_s,amb3_s,satis_2,length,numdat_2,attr7_2,sinc7_2,intel7_2,fun7_2,amb7_2,shar7_2,attr1_2,sinc1_2,intel1_2,fun1_2,amb1_2,shar1_2,attr4_2,sinc4_2,intel4_2,fun4_2,amb4_2,shar4_2,attr2_2,sinc2_2,intel2_2,fun2_2,amb2_2,shar2_2,attr3_2,sinc3_2,intel3_2,fun3_2,amb3_2,attr5_2,sinc5_2,intel5_2,fun5_2,amb5_2,you_call,them_cal,date_3,numdat_3,num_in_3,attr1_3,sinc1_3,intel1_3,fun1_3,amb1_3,shar1_3,attr7_3,sinc7_3,intel7_3,fun7_3,amb7_3,shar7_3,attr4_3,sinc4_3,intel4_3,fun4_3,amb4_3,shar4_3,attr2_3,sinc2_3,intel2_3,fun2_3,amb2_3,shar2_3,attr3_3,sinc3_3,intel3_3,fun3_3,amb3_3,attr5_3,sinc5_3,intel5_3,fun5_3,amb5_3
0,1,1.0,0,1,1,1,10,7,NaN,4,1,11.0,0,0.14,0,27.0,2.0,35.0,20.0,20.0,20.0,0.0,5.0,0,6.0,8.0,8.0,8.0,8.0,6.0,7.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,6.0,9.0,7.0,7.0,6.0,5.0,7.0,6.0,2.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
1,1,1.0,0,1,1,1,10,7,NaN,3,2,12.0,0,0.54,0,22.0,2.0,60.0,0.0,0.0,40.0,0.0,0.0,0,7.0,8.0,10.0,7.0,7.0,5.0,8.0,4.0,2.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,7.0,8.0,7.0,8.0,5.0,6.0,7.0,5.0,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
2,1,1.0,0,1,1,1,10,7,NaN,10,3,13.0,1,0.16,1,22.0,4.0,19.0,18.0,19.0,18.0,14.0,12.0,1,10.0,10.0,10.0,10.0,10.0,10.0,10.0,10.0,1.0,21.0,Law,1.0,NaN,NaN,NaN,4.0,2.0,4.0,Chicago,"60,521","69,487.00",2.0,7.0,1.0,lawyer,NaN,9.0,2.0,8.0,9.0,1.0,1.0,5.0,1.0,5.0,6.0,9.0,1.0,10.0,10.0,9.0,8.0,1.0,3.0,2.0,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,35.0,20.0,15.0,20.0,5.0,5.0,6.0,8.0,8.0,8.0,7.0,NaN,NaN,NaN,NaN,NaN,1,5.0,8.0,9.0,8.0,5.0,7.0,7.0,NaN,1.0,4.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,2.0,1.0,NaN,NaN,NaN,NaN,NaN,NaN,19.44,16.67,13.89,22.22,11.11,16.67,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,6.0,7.0,8.0,7.0,6.0,NaN,NaN,NaN,NaN,NaN,1.0,1.0,0.0,NaN,NaN,15.0,20.0,20.0,15.0,15.0,15.0,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,NaN,5.0,7.0,7.0,7.0,7.0,NaN,NaN,NaN,NaN,NaN
3,1,1.0,0,1,1,1,10,7,NaN,5,4,14.0,1,0.61,0,23.0,2.0,30.0,5.0,15.0,40.0,5.0,5.0,1,7.0,8.0,9.0,8.0,9

In [5]:
with pd.option_context("display.max_columns", None):
    display(df_tinder.isna().sum() / df_tinder.shape[0])

iid         0.000000
id          0.000119
gender      0.000000
idg         0.000000
condtn      0.000000
              ...   
attr5_3     0.759370
sinc5_3     0.759370
intel5_3    0.759370
fun5_3      0.759370
amb5_3      0.759370
Length: 195, dtype: float64

In [6]:
na_rates = (df_tinder.isna().sum() / df_tinder.shape[0]).reset_index()
na_rates.columns = ["column", "na_rates"]

na_rates

,column,na_rates
0,iid,0.000000
1,id,0.000119
2,gender,0.000000
3,idg,0.000000
4,condtn,0.000000
...,...,...
190,attr5_3,0.759370
191,sinc5_3,0.759370
192,intel5_3,0.759370
193,fun5_3,0.759370


In [7]:
with pd.option_context("display.max_rows", None):
    display(na_rates.sort_values("na_rates", ascending=False))

,column,na_rates
160,num_in_3,0.920267
159,numdat_3,0.821437
68,expnum,0.785152
123,sinc7_2,0.766651
126,amb7_2,0.766651
127,shar7_2,0.764383
122,attr7_2,0.763189
125,fun7_2,0.763189
124,intel7_2,0.763189
193,fun5_3,0.759370


### Dropping columns with too many missing values
Several columns contain more than 70–80% missing values, especially follow-up variables collected weeks after the event.  
They are not needed for the analysis, which focuses only on what happens during the speed-dating session itself.

In [8]:
high_na_cols = [
    "date_3","you_call","them_cal","numdat_3","num_in_3",
    "attr1_3","sinc1_3","intel1_3","fun1_3","amb1_3","shar1_3",
    "attr2_3","sinc2_3","intel2_3","fun2_3","amb2_3","shar2_3",
    "attr3_3","sinc3_3","intel3_3","fun3_3","amb3_3",
    "attr4_3","sinc4_3","intel4_3","fun4_3","amb4_3","shar4_3",
    "attr5_3","sinc5_3","intel5_3","fun5_3","amb5_3",
    "attr7_3","sinc7_3","intel7_3","fun7_3","amb7_3","shar7_3",
    "attr7_2","sinc7_2","intel7_2","fun7_2","amb7_2","shar7_2"
]

cols_to_drop = [c for c in high_na_cols if c in df_tinder.columns]

print(len(high_na_cols))
len(cols_to_drop)

45


45

In [9]:
df_clean = df_tinder.drop(columns=cols_to_drop).copy()

df_clean.shape

(8378, 150)

### Creating a core dataset

I keep only the columns that are useful for the analysis:  
IDs, event structure, decisions, ratings, demographics, and preferences.  
This creates a clean dataset while keeping the original data unchanged.

In [10]:
core_cols = [
    # IDs + event structure
    "iid", "id", "idg", "gender", "wave", "round",
    "position", "positin1", "order", "partner", "pid", "condtn",

    # Decisions (participant and partner)
    "match", "dec", "dec_o", "int_corr", "samerace",

    # Demographics
    "age", "age_o", "race", "race_o",
    "field", "field_cd", "mn_sat", "tuition", "income",
    "goal", "date", "go_out", "imprace", "imprelig",
    "from", "zipcode",

    # Participant (partner ratings during the event)
    "attr", "sinc", "intel", "fun", "amb", "shar",
    "like", "prob", "met",

    # Partner (participant ratings during the event)
    "attr_o", "sinc_o", "intel_o", "fun_o", "amb_o", "shar_o",
    "like_o", "prob_o", "met_o",

    # Preferences declared before the event (Time 1)
    "attr1_1", "sinc1_1", "intel1_1", "fun1_1", "amb1_1", "shar1_1",

    # What participant thinks the opposite sex wants
    "attr2_1", "sinc2_1", "intel2_1", "fun2_1", "amb2_1", "shar2_1",

    # Self-assessment
    "attr3_1", "sinc3_1", "intel3_1", "fun3_1", "amb3_1",

    # How participant thinks others see them
    "attr5_1", "sinc5_1", "intel5_1", "fun5_1", "amb5_1",

    # Expectations before the event
    "exphappy", "expnum", "match_es"
]

df_core = df_clean[core_cols].copy()

df_core.info()
df_core.shape


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8378 entries, 0 to 8377
Data columns (total 76 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   iid       8378 non-null   int64  
 1   id        8377 non-null   float64
 2   idg       8378 non-null   int64  
 3   gender    8378 non-null   int64  
 4   wave      8378 non-null   int64  
 5   round     8378 non-null   int64  
 6   position  8378 non-null   int64  
 7   positin1  6532 non-null   float64
 8   order     8378 non-null   int64  
 9   partner   8378 non-null   int64  
 10  pid       8368 non-null   float64
 11  condtn    8378 non-null   int64  
 12  match     8378 non-null   int64  
 13  dec       8378 non-null   int64  
 14  dec_o     8378 non-null   int64  
 15  int_corr  8220 non-null   float64
 16  samerace  8378 non-null   int64  
 17  age       8283 non-null   float64
 18  age_o     8274 non-null   float64
 19  race      8315 non-null   float64
 20  race_o    8305 non-null   floa

(8378, 76)

In [11]:
rename_dict = {
    "iid": "subject_id",
    "id": "subject_id_wave",
    "idg": "subject_id_gender", 
    "wave": "event_wave",
    "round": "total_dates_wave",
    "position": "table_position",
    "positin1": "start_position",
    "order": "date_order",
    "partner": "partner_wave_id",
    "pid": "partner_subject_id",
    "condtn": "choice_condition",
    "match": "mutual_match",               
    "dec": "participant_decision",           
    "dec_o": "partner_decision",             
    "int_corr": "interest_correlation",
    "samerace": "same_race",
    "age": "age",
    "age_o": "partner_age",
    "race": "race",
    "race_o": "partner_race",
    "field": "field_study",
    "field_cd": "field_study_code",
    "mn_sat": "undergrad_sat_score",
    "tuition": "undergrad_tuition",
    "income": "income_zipcode",
    "goal": "event_goal",
    "date": "date_frequency",
    "go_out": "going_out_frequency",
    "imprace": "importance_same_race",
    "imprelig": "importance_same_religion",
    "from": "origin_region",
    "zipcode": "origin_zipcode",
    "attr": "p_rating_attr",
    "sinc": "p_rating_sinc",
    "intel": "p_rating_intel",
    "fun": "p_rating_fun",
    "amb": "p_rating_amb",
    "shar": "p_rating_shared_interests",
    "like": "p_rating_like",
    "prob": "p_probs_partner_yes",
    "met": "p_met_before",
    "attr_o": "o_rating_attr",
    "sinc_o": "o_rating_sinc",
    "intel_o": "o_rating_intel",
    "fun_o": "o_rating_fun",
    "amb_o": "o_rating_amb",
    "shar_o": "o_rating_shared_interests",
    "like_o": "o_rating_like",
    "prob_o": "o_probs_participant_yes",
    "met_o": "o_met_before",
    "attr1_1": "pref_attr",
    "sinc1_1": "pref_sinc",
    "intel1_1": "pref_intel",
    "fun1_1": "pref_fun",
    "amb1_1": "pref_amb",
    "shar1_1": "pref_shared_interests",
    "attr2_1": "belief_oppsex_attr",
    "sinc2_1": "belief_oppsex_sinc",
    "intel2_1": "belief_oppsex_intel",
    "fun2_1": "belief_oppsex_fun",
    "amb2_1": "belief_oppsex_amb",
    "shar2_1": "belief_oppsex_shared_interests",
    "attr3_1": "self_attr",
    "sinc3_1": "self_sinc",
    "intel3_1": "self_intel",
    "fun3_1": "self_fun",
    "amb3_1": "self_amb",
    "attr5_1": "others_view_attr",
    "sinc5_1": "others_view_sinc",
    "intel5_1": "others_view_intel",
    "fun5_1": "others_view_fun",
    "amb5_1": "others_view_amb",
    "exphappy": "expected_happiness",
    "expnum": "expected_num_interested",
    "match_es": "expected_num_matches"
}

df_core = df_core.rename(columns=rename_dict)

df_core.head()

,subject_id,subject_id_wave,subject_id_gender,gender,event_wave,total_dates_wave,table_position,start_position,date_order,partner_wave_id,...,self_fun,self_amb,others_view_attr,others_view_sinc,others_view_intel,others_view_fun,others_view_amb,expected_happiness,expected_num_interested,expected_num_matches
0,1,1.0,1,0,1,10,7,NaN,4,1,...,8.0,7.0,NaN,NaN,NaN,NaN,NaN,3.0,2.0,4.0
1,1,1.0,1,0,1,10,7,NaN,3,2,...,8.0,7.0,NaN,NaN,NaN,NaN,NaN,3.0,2.0,4.0
2,1,1.0,1,0,1,10,7,NaN,10,3,...,8.0,7.0,NaN,NaN,NaN,NaN,NaN,3.0,2.0,4.0
3,1,1.0,1,0,1,10,7,NaN,5,4,...,8.0,7.0,NaN,NaN,NaN,NaN,NaN,3.0,2.0,4.0
4,1,1.0,1,0,1,10,7,NaN,7,5,...,8.0,7.0,NaN,NaN,NaN,NaN,NaN,3.0,2.0,4.0


What are the least desirable attributes in a male partner? Does this differ for female partners?

In [12]:
(df_core["gender"].isna().sum() / df_core.shape[0])

np.float64(0.0)

In [13]:
df_core["gender"].value_counts()

gender
1    4194
0    4184
Name: count, dtype: int64

In [14]:
df_gender = (
    df_core["gender"]
    .value_counts()
    .reset_index()
)

px.pie(df_gender, values="count", names="gender", title="Gender distribution")

In [15]:
df_core[df_core["gender"] == 1]

,subject_id,subject_id_wave,subject_id_gender,gender,event_wave,total_dates_wave,table_position,start_position,date_order,partner_wave_id,...,self_fun,self_amb,others_view_attr,others_view_sinc,others_view_intel,others_view_fun,others_view_amb,expected_happiness,expected_num_interested,expected_num_matches
100,11,1.0,2,1,1,10,7,NaN,4,1,...,7.0,5.0,NaN,NaN,NaN,NaN,NaN,7.0,3.0,0.0
101,11,1.0,2,1,1,10,3,NaN,10,2,...,7.0,5.0,NaN,NaN,NaN,NaN,NaN,7.0,3.0,0.0
102,11,1.0,2,1,1,10,9,NaN,6,3,...,7.0,5.0,NaN,NaN,NaN,NaN,NaN,7.0,3.0,0.0
103,11,1.0,2,1,1,10,6,NaN,3,4,...,7.0,5.0,NaN,NaN,NaN,NaN,NaN,7.0,3.0,0.0
104,11,1.0,2,1,1,10,4,NaN,1,5,...,7.0,5.0,NaN,NaN,NaN,NaN,NaN,7.0,3.0,0.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
8373,552,22.0,44,1,21,22,14,10.0,5,18,...,6.0,7.0,9.0,7.0,10.0,5.0,9.0,10.0,NaN,3.0
8374,552,22.0,44,1,21,22,13,10.0,4,19,...,6.0,7.0,9.0,7.0,10.0,5.0,9.0,10.0,NaN,3.0
8375,552,22.0,44,1,21,22,19,10.0,10,20,...,6.0,7.0,9.0,7.0,10.0,5.0,9.0,10.0,NaN,3.0
8376,552,22.0,44,1,21,22,3,10.0,16,21,...,6.0,7.0,9.0,7.0,10.0,5.0,9.0,10.0,NaN,3.0


In [16]:
df_male = df_core[df_core["gender"] == 1]
df_female = df_core[df_core["gender"] == 0]

print(df_male.shape)
df_female.shape

(4194, 76)


(4184, 76)

In [17]:
pref_cols = [
    "pref_attr", "pref_sinc", "pref_intel",
    "pref_fun", "pref_amb", "pref_shared_interests"
]

male_prefs = df_male[pref_cols].mean().round(2)
female_prefs = df_female[pref_cols].mean().round(2)

print(male_prefs)
female_prefs

pref_attr                26.92
pref_sinc                16.50
pref_intel               19.54
pref_fun                 17.76
pref_amb                  8.55
pref_shared_interests    11.00
dtype: float64


pref_attr                18.06
pref_sinc                18.31
pref_intel               21.00
pref_fun                 17.15
pref_amb                 12.83
pref_shared_interests    12.70
dtype: float64

In [18]:
df_prefs = pd.DataFrame({
    "attribute": pref_cols,
    "male_mean": male_prefs.values,
    "female_mean": female_prefs.values
})

df_prefs

,attribute,male_mean,female_mean
0,pref_attr,26.92,18.06
1,pref_sinc,16.50,18.31
2,pref_intel,19.54,21.00
3,pref_fun,17.76,17.15
4,pref_amb,8.55,12.83
5,pref_shared_interests,11.00,12.70


In [19]:
df_long = df_prefs.melt(
    id_vars="attribute",
    value_vars=["male_mean", "female_mean"],
    var_name="gender",
    value_name="importance"
)

df_long

,attribute,gender,importance
0,pref_attr,male_mean,26.92
1,pref_sinc,male_mean,16.50
2,pref_intel,male_mean,19.54
3,pref_fun,male_mean,17.76
4,pref_amb,male_mean,8.55
5,pref_shared_interests,male_mean,11.00
6,pref_attr,female_mean,18.06
7,pref_sinc,female_mean,18.31
8,pref_intel,female_mean,21.00
9,pref_fun,female_mean,17.15


In [20]:
px.bar(
    df_long, 
    x="attribute", 
    y="importance", 
    color="gender",
    barmode="group",
    title="Attribute importance declared by gender"
)

Men overwhelmingly prioritize physical attractiveness, making it the most important trait(27%).  
Women place more weight on intelligence and sincerity, with attractiveness important but not dominant (18%).  
Both genders value “fun” and shared interests, but ambition is significantly more important for women than for men.

In [21]:
female_prefs_sorted = df_long[df_long["gender"] == "female_mean"].sort_values("importance")

female_prefs_sorted

,attribute,gender,importance
11,pref_shared_interests,female_mean,12.70
10,pref_amb,female_mean,12.83
9,pref_fun,female_mean,17.15
6,pref_attr,female_mean,18.06
7,pref_sinc,female_mean,18.31
8,pref_intel,female_mean,21.00


In [22]:
px.bar(
    female_prefs_sorted, 
    x="attribute", 
    y="importance",
    title="Female preferences for male partner attributes" )

Women rate “shared interests” and “ambition” as the least important attributes in a male partner, compared to intelligence, sincerity and attractiveness.

In [23]:
male_prefs_sorted = df_long[df_long["gender"] == "male_mean"].sort_values("importance")

male_prefs_sorted

,attribute,gender,importance
4,pref_amb,male_mean,8.55
5,pref_shared_interests,male_mean,11.00
1,pref_sinc,male_mean,16.50
3,pref_fun,male_mean,17.76
2,pref_intel,male_mean,19.54
0,pref_attr,male_mean,26.92


In [24]:
px.bar(male_prefs_sorted, x="attribute", y="importance", title="Male preferences for female partner attributes")

Men consider “ambition” and “shared interests” as the least important attributes in a female partner, far below attractiveness, intelligence, fun and sincerity.

Both men and women rank ambition and shared interests as the least important attributes in a potential partner.  
What really differs is how much importance they assign to the other traits: men place much more weight on attractiveness, while women prioritise intelligence and sincerity.

How important do people think attractiveness is in potential mate selection vs. its real impact?

In [25]:
pref_cols = [
    "pref_attr", "p_rating_attr",
    "participant_decision"
]

df_attr_male = df_male[pref_cols].mean().round(2)
df_attr_female = df_female[pref_cols].mean().round(2)

In [26]:
print(df_attr_male)
df_attr_female

pref_attr               26.92
p_rating_attr            6.46
participant_decision     0.47
dtype: float64


pref_attr               18.06
p_rating_attr            5.92
participant_decision     0.37
dtype: float64

In [27]:
df_attr = pd.DataFrame({
    "attribute": pref_cols,
    "male_mean": df_attr_male.values,
    "female_mean": df_attr_female.values
})

df_attr

,attribute,male_mean,female_mean
0,pref_attr,26.92,18.06
1,p_rating_attr,6.46,5.92
2,participant_decision,0.47,0.37


In [28]:
df_attr = df_attr.melt(
    id_vars="attribute",
    value_vars=["male_mean", "female_mean"],
    var_name="gender",
    value_name="importance"
)

df_attr

,attribute,gender,importance
0,pref_attr,male_mean,26.92
1,p_rating_attr,male_mean,6.46
2,participant_decision,male_mean,0.47
3,pref_attr,female_mean,18.06
4,p_rating_attr,female_mean,5.92
5,participant_decision,female_mean,0.37


In [29]:
px.bar(
    df_attr, 
    x="attribute", 
    y="importance",
    color="gender",
    barmode="group",
    title="Perceived vs actual role of attractiveness by gender")

Men report that attractiveness is much more important to them than women do, but in practice both genders give similar attractiveness ratings to their partners. Men place high declared importance on attractiveness, yet they show a higher acceptance rate than women. This suggests that their reported preferences are not perfectly aligned with their actual decision behavior.

Real impact of attractiveness on decisions (YES vs NO)

In [30]:
df_male_yes = df_male[df_male["participant_decision"] == 1]
df_male_no = df_male[df_male["participant_decision"] == 0]

df_female_yes = df_female[df_female["participant_decision"] == 1]
df_female_no = df_female[df_female["participant_decision"] == 0]

In [31]:
male_mean_yes = df_male_yes["p_rating_attr"].mean().round(2)
male_mean_no = df_male_no["p_rating_attr"].mean().round(2)

female_mean_yes = df_female_yes["p_rating_attr"].mean().round(2)
female_mean_no = df_female_no["p_rating_attr"].mean().round(2)

In [32]:
df_attri_mean = pd.DataFrame({
    "group":["male_yes", "male_no", "female_yes", "female_no"],
    "mean_p_rating_attr": [
        male_mean_yes,
        male_mean_no,
        female_mean_yes,
        female_mean_no
    ]
})

df_attri_mean

,group,mean_p_rating_attr
0,male_yes,7.45
1,male_no,5.53
2,female_yes,7.07
3,female_no,5.23


In [33]:
px.bar(
    df_attri_mean,
    x="group",
    y="mean_p_rating_attr",
    title="Actual Impact of Attractiveness on the Yes/No Decision",
    labels={"mean_p_rating_attr": "Mean participant attractiveness rating (1–10)"},
    color="group"
)

People are significantly more likely to say “yes” when they rate their partner as more attractive.  
Men say yes when attractiveness reaches 7.45 on average, compared to 5.53 for their no-decisions.  
Women show the same pattern: they say yes at around 7.07 vs 5.23 for no-decisions.  
This confirms that attractiveness has a strong real impact, for both genders.

Are shared interests more important than a shared racial background?  
We also include religious background in the perceived-importance comparison, since the dataset does not contain any variable indicating whether two participants actually share the same religion.

In [34]:
df_core.columns

Index(['subject_id', 'subject_id_wave', 'subject_id_gender', 'gender',
       'event_wave', 'total_dates_wave', 'table_position', 'start_position',
       'date_order', 'partner_wave_id', 'partner_subject_id',
       'choice_condition', 'mutual_match', 'participant_decision',
       'partner_decision', 'interest_correlation', 'same_race', 'age',
       'partner_age', 'race', 'partner_race', 'field_study',
       'field_study_code', 'undergrad_sat_score', 'undergrad_tuition',
       'income_zipcode', 'event_goal', 'date_frequency', 'going_out_frequency',
       'importance_same_race', 'importance_same_religion', 'origin_region',
       'origin_zipcode', 'p_rating_attr', 'p_rating_sinc', 'p_rating_intel',
       'p_rating_fun', 'p_rating_amb', 'p_rating_shared_interests',
       'p_rating_like', 'p_probs_partner_yes', 'p_met_before', 'o_rating_attr',
       'o_rating_sinc', 'o_rating_intel', 'o_rating_fun', 'o_rating_amb',
       'o_rating_shared_interests', 'o_rating_like', 'o_probs_par

In [35]:
importance_cols = [
    "importance_same_race",
    "importance_same_religion", "pref_shared_interests"
]

df_importance = df_core[importance_cols].mean().round(2)

df_importance

importance_same_race         3.78
importance_same_religion     3.65
pref_shared_interests       11.85
dtype: float64

In [36]:
df_importance = pd.DataFrame({
    "importance": importance_cols,
    "mean": df_importance.values.flatten()
}).sort_values("mean")

df_importance

,importance,mean
1,importance_same_religion,3.65
0,importance_same_race,3.78
2,pref_shared_interests,11.85


In [37]:
px.bar(
    df_importance, 
    x="importance", 
    y="mean", 
    title="Perceived Importance of Race, Religion and Shared Interests")

Participants place more subjective importance on shared interests than on racial or religious similarity.
However, the three measures are collected on different scales, so their absolute values cannot be compared directly.

In [38]:
df_yes = df_core[df_core["participant_decision"] == 1]
df_no = df_core[df_core["participant_decision"] == 0]

In [39]:
interests_cols = [
    "same_race",
    "p_rating_shared_interests"
]

df_yes_interest = df_yes[interests_cols]
df_no_interest = df_no[interests_cols]

df_no_interest.head()

,same_race,p_rating_shared_interests
5,0,4.0
7,0,6.0
10,1,3.0
11,1,6.0
12,0,4.0


In [40]:
df_yes_interest_mean = df_yes_interest.mean().round(2)
df_no_interest_mean = df_no_interest.mean().round(2)

df_yes_interest_mean

same_race                    0.41
p_rating_shared_interests    6.46
dtype: float64

In [41]:
df_interest = pd.DataFrame({
    "interest": interests_cols,
    "yes": df_yes_interest_mean.values,
    "no": df_no_interest_mean.values
})

df_interest

,interest,yes,no
0,same_race,0.41,0.39
1,p_rating_shared_interests,6.46,4.72


In [42]:
df_long = df_interest.melt(
    id_vars="interest",
    value_vars=["yes", "no"],
    var_name="decision",
    value_name="mean_value"
)

df_long

,interest,decision,mean_value
0,same_race,yes,0.41
1,p_rating_shared_interests,yes,6.46
2,same_race,no,0.39
3,p_rating_shared_interests,no,4.72


In [43]:
px.bar(
    df_long,
    x="interest",
    y="mean_value",
    barmode="group",
    color="decision",
    title="Real impact of shared race vs shared interests on the decision"
)

The two variables are not measured on the same scale (proportion for race vs a 1–10 rating for shared interests).
Shared interests have a strong real impact on the decision, while racial similarity does not.
In the plot, we see that the average shared-interests rating is much higher for “yes” decisions (6.46) than for “no” decisions (4.72), showing a clear positive effect.
In contrast, the proportion of same-race pairs is almost identical for yes and no (0.41 vs 0.39), which means that sharing the same racial background has almost no influence on whether someone says yes.

Can people accurately predict their own perceived value in the dating market?

In [46]:
self_cols = [
    "self_attr",
    "self_sinc",
    "self_intel",
    "self_fun",
    "self_amb"
]

partner_cols = [
    "o_rating_attr",
    "o_rating_sinc",
    "o_rating_intel",
    "o_rating_fun",
    "o_rating_amb"
]

cols_self_patner = self_cols + partner_cols
df_self_partner = df_core[cols_self_patner]

df_self_partner.head()

,self_attr,self_sinc,self_intel,self_fun,self_amb,o_rating_attr,o_rating_sinc,o_rating_intel,o_rating_fun,o_rating_amb
0,6.0,8.0,8.0,8.0,7.0,6.0,8.0,8.0,8.0,8.0
1,6.0,8.0,8.0,8.0,7.0,7.0,8.0,10.0,7.0,7.0
2,6.0,8.0,8.0,8.0,7.0,10.0,10.0,10.0,10.0,10.0
3,6.0,8.0,8.0,8.0,7.0,7.0,8.0,9.0,8.0,9.0
4,6.0,8.0,8.0,8.0,7.0,8.0,7.0,9.0,6.0,9.0


In [47]:
self_mean = df_self_partner[self_cols].mean().round(2)
partner_mean = df_self_partner[partner_cols].mean().round(2)

df_self_vs_partner = pd.DataFrame({
    "attribute": ["Attractiveness", "Sincerity", "Intelligence", "Fun", "Ambition"],
    "self_rating": self_mean.values,
    "partner_rating": partner_mean.values
})

df_self_vs_partner

,attribute,self_rating,partner_rating
0,Attractiveness,7.08,6.19
1,Sincerity,8.29,7.18
2,Intelligence,8.40,7.37
3,Fun,7.70,6.40
4,Ambition,7.58,6.78


In [49]:
df_long = df_self_vs_partner.melt(
    id_vars="attribute",
    value_vars=["self_rating", "partner_rating"],
    var_name="rating_type",
    value_name="score"
)

df_long

,attribute,rating_type,score
0,Attractiveness,self_rating,7.08
1,Sincerity,self_rating,8.29
2,Intelligence,self_rating,8.40
3,Fun,self_rating,7.70
4,Ambition,self_rating,7.58
5,Attractiveness,partner_rating,6.19
6,Sincerity,partner_rating,7.18
7,Intelligence,partner_rating,7.37
8,Fun,partner_rating,6.40
9,Ambition,partner_rating,6.78


In [51]:
px.bar(
    df_long,
    x="attribute",
    y="score",
    color="rating_type",
    barmode="group",
    title="Self-Rating vs Partner-Rating Across Traits"
)

People are not very accurate in predicting their own perceived value in the dating market.
On average, participants give themselves higher ratings than their partners give them, with a consistent gap across all traits.

To understand whether people can truly predict how others will perceive them, we also need to measure the relationship between self-ratings and partner-ratings.

In [52]:
attributes = ["attr", "sinc", "intel", "fun", "amb"]

correlation_results = []

for attribute in attributes:
    self_col = f"self_{attribute}"
    partner_col = f"o_rating_{attribute}"

    corr_value = df_core[self_col].corr(df_core[partner_col])

    correlation_results.append({
        "attribute": attribute,
        "correlation": round(corr_value, 3)
    })

df_corr = pd.DataFrame(correlation_results)
df_corr

,attribute,correlation
0,attr,0.175
1,sinc,0.001
2,intel,0.023
3,fun,0.146
4,amb,0.056


All correlations between self-ratings and partner-ratings are extremely low.
A perfect prediction would show a correlation close to 1, but the values observed here range from 0.00 to 0.18, which indicates almost no relationship between how people see themselves and how their dates actually perceive them.

People are not accurate at predicting their own perceived value in the dating market.
Self-ratings and partner-ratings correlate extremely weakly (0.00–0.18), showing that how participants evaluate themselves has almost no relationship with how others evaluate them during the speed dates.
Combined with the systematic self-overestimation observed earlier, these results show that people have a poor intuition about how desirable they actually are.

In terms of getting a second date, is it better to be someone's first speed date of the night or their last?

In [73]:
cols_to_keep = ["date_order", "partner_decision"]
df_order = df_core[cols_to_keep]
df_order

,date_order,partner_decision
0,4,0
1,3,0
2,10,1
3,5,1
4,7,1
...,...,...
8373,5,1
8374,4,0
8375,10,0
8376,16,1


In [75]:
df_order = (
    df_order
    .groupby("date_order")
    .mean()
    .round(2)
    .reset_index()
    .rename(columns={"partner_decision": "yes_rate"})
)
df_order

,date_order,yes_rate
0,1,0.49
1,2,0.40
2,3,0.41
3,4,0.44
4,5,0.44
5,6,0.43
6,7,0.41
7,8,0.41
8,9,0.45
9,10,0.41


In [79]:
px.line(
    df_order,
    x="date_order",
    y="yes_rate",
    markers=True,
    title="Yes rate by date order"
)

Participants are more likely to receive a “yes” at the beginning of the evening than at the end.
The first position shows the highest yes-rate, the middle positions remain relatively stable with moderate values, and although some late positions rise again, the final ones tend to show the lowest rates.
This suggests that the order in which someone is met can influence the outcome: early encounters generally perform better than later ones.

Overall, several factors shape whether participants want a second date.  
First, men and women differ in the attributes they say they value, but in practice, attractiveness and shared interests are the strongest real predictors of a “yes”. Shared race and religion, although rated as somewhat important by a few participants, show almost no measurable impact on the final decision.  
Participants also tend to overestimate their own desirability: self-ratings are consistently higher than partner ratings, and the correlation between the two is extremely low, indicating that people are not accurate in predicting how they will be perceived.  
The order of the encounters also matters: early positions in the evening receive the highest yes-rates, middle positions remain stable, and the very last positions tend to receive the lowest ones.  
Taken together, the results show that immediate impressions such as appearance, perceived connection, and position in the sequence play a far greater role in speed-dating outcomes than demographic similarities or self-perception.